calico_node/tests/st/config/test_felix_config.py

```diff
@@ -0,0 +1,410 @@
+    def test_failsafe_outbound(self):
+        """
+        Test failsafe outbound configuration.
+        """
+        # We need to put 2379 into random because felix need to connect to etcd.
+        random = {"udp": [70], "tcp": [2379, 3000, 5000]}
+        default = {"udp": [53, 67], "tcp": [179, 2379, 2380, 6666, 6667]}
+
+        # Test random and default ports can be accessed with no host endpoints setup.
+        self.run_failsafe(default, True)
+        self.run_failsafe(random, True)
+
+        # With host endpoints setup. Default ports can be accessed.
+        # But random ports can not.
+        self.add_host_iface(self.host2_hostname, self.host2.ip)
+        self.run_failsafe(default, True)
+        self.run_failsafe(random, False)
+
+        # Update felix config and check again.
+        self.add_felix_failsafe_config("default", random, "outbound")
+        self.run_failsafe(default, False)
+        self.run_failsafe(random, True)
+
+        # Put default back for host2.
+        self.add_felix_failsafe_config("node.%s" % self.host2_hostname, default, "outbound")
+        self.run_failsafe(default, True)
+        self.run_failsafe(random, False)
+
+        # Put default back for all.
+        self.add_felix_failsafe_config("default", default, "outbound")
+        return
+
+    @handle_failure
+    def test_log_file(self):
+        """
+        Test Log file configuration.
+        """
+        conf_name = "node.%s" % self.host1_hostname
+        default = "/var/log/calico/felix/current"
+
+        # Check default log file is good.
+        retry_until_success(self._get_check_file_func(True, default, "INFO"),
+                            retries=3)
+        retry_until_success(self._get_check_file_func(False, default, "DEBUG"),
+                            retries=3)
+
+        # Change severity to DEBUG.
+        self.add_felix_config(conf_name, {'LogSeverityFile': 'DEBUG'})
+        retry_until_success(self._get_check_file_func(True, default, "DEBUG"),
+                            retries=3)
+
+        # Change log file and severity to FATAL.
+        new_path = "/var/log/calico/st_test0"
+        self.add_felix_config(conf_name, {'logFilePath': new_path, 'LogSeverityFile': 'FATAL'})
+        self.restart_felix_no_config_file(self.host1)
+
+        # Dont expect FATAL or WARNING message from new log file.
+        retry_until_success(self._get_check_file_func(False, new_path, "WARNING"),
+                            retries=3)
+        retry_until_success(self._get_check_file_func(False, new_path, "FATAL"),
+                            retries=3)
+
+        # Send a termination signal. We should see FATAL but not WARNING.
+        self.restart_felix_no_config_file(self.host1)
+        retry_until_success(self._get_check_file_func(False, new_path, "WARNING"),
+                            retries=3)
+        retry_until_success(self._get_check_file_func(True, new_path, "FATAL"),
+                            retries=3)
+
```

Comment:
Is it worth checking we don't have INFO or DEBUG too?

Reply:
Yes.